src/Files.App/Filesystem/StorageFileHelpers/StorageFileExtensions.cs

```diff
@@ -242,16 +244,15 @@ public static string GetResolvedPath(string path, bool isFtp)
 				}
 			}
 
-			if (parentFolder is not null && !Path.IsPathRooted(value) && !ShellStorageFolder.IsShellPath(value)) // "::{" not a valid root
-			{
-				// Relative path
-				var fullPath = Path.GetFullPath(Path.Combine(parentFolder.Path, value));
-				return new StorageFolderWithPath(await BaseStorageFolder.GetFolderFromPathAsync(fullPath));
-			}
-			else
-			{
-				return new StorageFolderWithPath(await BaseStorageFolder.GetFolderFromPathAsync(value));
-			}
+			var fullPath = (parentFolder is not null && !FtpHelpers.IsFtpPath(value) && !Path.IsPathRooted(value) && !ShellStorageFolder.IsShellPath(value)) // "::{" not a valid root
+				? Path.GetFullPath(Path.Combine(parentFolder.Path, value)) // Relative path
+				: value;
+			var item = await BaseStorageFolder.GetFolderFromPathAsync(fullPath);
+
+			if (parentFolder is not null && parentFolder.Item is IPasswordProtectedItem ppis && item is IPasswordProtectedItem ppid)
+				ppid.Credentials = ppis.Credentials;
+
+			return new StorageFolderWithPath(item);
```

Comment:
Repetition with l.193 to 201, it can be refactored into one method.

Reply:
Note that one method handles a storage file, the other a storage folder (GetFileFromPathAsync() vs GetFolderFromPathAsync()).
The merged method would look something like this?

```cs
public async static Task<StorageFileWithPath> DangerousGetFileWithPathFromPathAsync
	(string value, StorageFolderWithPath rootFolder = null, StorageFolderWithPath parentFolder = null)
	...
	return new StorageFileWithPath(await Foo<BaseStorageFile>(value, parentFolder));
}

public static async Task<T> Foo<T>(string value, StorageFolderWithPath? parentFolder)
{
	var fullPath = (parentFolder is not null && !FtpHelpers.IsFtpPath(value) && !Path.IsPathRooted(value) && !ShellStorageFolder.IsShellPath(value)) // "::{" not a valid root
		? Path.GetFullPath(Path.Combine(parentFolder.Path, value)) // Relative path
		: value;

	var type = typeof(T);
	IStorageItem item = type switch
	{
		Type _ when type == typeof(BaseStorageFile) => await BaseStorageFile.GetFileFromPathAsync(fullPath),
		Type _ when type == typeof(BaseStorageFolder) => await BaseStorageFolder.GetFolderFromPathAsync(fullPath),
		_ => throw new ArgumentException($"Invalid type param: {type.Name}")
	};
	
	if (parentFolder is not null && parentFolder.Item is IPasswordProtectedItem ppis && item is IPasswordProtectedItem ppid)
		ppid.Credentials = ppis.Credentials;

	return (T)item;
}
```